contracts/V1/StateChannelDiamondProxy/DisputeFraudProofFacet.sol

```diff
@@ -377,32 +386,194 @@ contract DisputeFraudProofFacet is StateChannelCommon {
         returns (address)
     {
         TimeoutCalldataPosted memory timeoutCalldataPostedProof = abi.decode(encodedFraudProof, (TimeoutCalldataPosted));
-        Block memory postedBlock = timeoutCalldataPostedProof.postedBlock;
+        SignedBlock memory postedBlock = timeoutCalldataPostedProof.postedBlock;
+        Block memory _block = abi.decode(postedBlock.encodedBlock, (Block));
+        StateSnapshot memory latestStateSnapshot = timeoutCalldataPostedProof.auditingData.latestStateSnapshot;
+
+        // Requires correct auditing data
+        require(
+            _checkDisputeAuditingDataCommitment(dispute, timeoutCalldataPostedProof.auditingData),
+            ErrorAuditingDataHashMismatch()
+        );
 
         // Check channelId
-        if (!_areDisputeAndBlockSameChannel(dispute, postedBlock)) revert();
+        if (!_areDisputeAndBlockSameChannel(dispute, _block)) return address(0); // the calling context may decide to slash the caller
 
         // Check forkId
-        if (!_areDisputeAndBlockSameFork(dispute, postedBlock)) revert();
+        if (!_areDisputeAndBlockSameFork(dispute, _block)) return address(0); // the calling context may decide to slash the caller
 
         // Check timeout == postedBlock
-        if (dispute.input.timeout.blockHeight != _getBlockHeight(postedBlock)) revert();
+        if (dispute.input.timeout.blockHeight != _getBlockHeight(_block)) return address(0); // the calling context may decide to slash the caller
 
         // Check timeout participant == block author
-        if (dispute.input.timeout.participant != _getBlockAuthor(postedBlock)) revert();
+        if (dispute.input.timeout.participant != _getBlockAuthor(_block)) return address(0); // the calling context may decide to slash the caller
 
         // Check block calldata posted
-        (bool isFound,) = getBlockCallDataCommitment(
+        (bool isFound, bytes32 commitment) = getBlockCallDataCommitment(
             _getDisputeChannel(dispute),
             _getDisputeFork(dispute),
             dispute.input.timeout.blockHeight,
             dispute.input.timeout.participant
         );
-        if (!isFound) revert();
+        if (!isFound) return address(0); // the calling context may decide to slash the caller
+        bytes32 _commitment = keccak256(abi.encode(postedBlock, timeoutCalldataPostedProof.onChainTimestamp));
```

Comment:
unclear nameing
I get what you are doing here
but comparing commitment to _commitment is not very revealing